src/main/java/org/example/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+package org.example.bookstore.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.example.bookstore.dto.order.ChangeOrderStatusRequestDto;
+import org.example.bookstore.dto.order.CreateOrderRequestDto;
+import org.example.bookstore.dto.order.OrderDto;
+import org.example.bookstore.dto.order.item.OrderItemDto;
+import org.example.bookstore.exception.DataProcessingException;
+import org.example.bookstore.exception.EntityNotFoundException;
+import org.example.bookstore.mapper.ItemCartOrderMapper;
+import org.example.bookstore.mapper.OrderItemMapper;
+import org.example.bookstore.mapper.OrderMapper;
+import org.example.bookstore.model.CartItem;
+import org.example.bookstore.model.Order;
+import org.example.bookstore.model.OrderItem;
+import org.example.bookstore.model.ShoppingCart;
+import org.example.bookstore.model.User;
+import org.example.bookstore.model.constant.Status;
+import org.example.bookstore.repository.order.OrderRepository;
+import org.example.bookstore.repository.order.item.OrderItemRepository;
+import org.example.bookstore.repository.shopping.cart.ShoppingCartRepository;
+import org.example.bookstore.service.OrderService;
+import org.example.bookstore.service.ShoppingCartService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService shoppingCartService;
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final ItemCartOrderMapper itemCartOrderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public List<OrderDto> findAllByUserId(Long userId) {
+        return orderMapper.toDto(orderRepository.findAllByUserId(userId));
+    }
+
+    @Override
+    public OrderDto createOrder(CreateOrderRequestDto createOrderRequestDto, Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId).orElseThrow(
+                () -> new EntityNotFoundException("Can`t find shopping cart with user id " + userId)
+        );
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new DataProcessingException("Can`t create order, because shopping cart of user "
+                    + "with id " + userId + " is empty");
+        }
+        Order order = createBlankOrder(createOrderRequestDto, userId);
```

Comment:
Other option of implementation

```suggestion
        Order order = orderMapper.cartToOrder(cart, orderDto.shippingAddress());
        shoppingCart.clearCart();
        return....
```

Where in OrderMapper

```
@Mapping(target = "id", ignore = true)
@Mapping(target = "total", source = "cart.cartItems", qualifiedByName = "total")
@Mapping(target = "orderItems", source = "cart.cartItems")
Order cartToOrder(ShoppingCart cart, String shippingAddress);
```

```
@AfterMapping
    default void updateOrder(@MappingTarget Order order) {
        order.getOrderItems().forEach(oi -> oi.setOrder(order));
    }

    @Named("total")
    default BigDecimal getTotal(Set<CartItem> cartItems) {
        return cartItems.stream()
                .map(i -> i.getBook().getPrice().multiply(BigDecimal.valueOf(i.getQuantity())))
                .reduce(BigDecimal.ZERO, BigDecimal::add);
    }

```

and in OrderItemMapper

```
@Mapping(target = "id", ignore = true)
@Mapping(target = "price", source = "book.price")
OrderItem cartItemToOrderItem(CartItem cartItem);
```